product/src/main/java/com/yas/product/service/ProductService.java

```diff
@@ -184,15 +184,15 @@ public ProductGetDetailVm updateProduct(long productId, ProductPutVm productPutV
             throw new BadRequestException(String.format("Slug %s is duplicated", productPutVm.slug()));
         }
 
+        product.setBrand(null);
         if (productPutVm.brandId() != null) {
             Brand brand = brandRepository.findById(productPutVm.brandId()).
                     orElseThrow(() -> new NotFoundException(String.format("Brand %s is not found", productPutVm.brandId())));
             product.setBrand(brand);
         }
-
+        productCategoryRepository.deleteAll(product.getProductCategories());
+        product.setProductCategories(null);
```

Comment:
Delete all and re-create for every update request seem not a good idea. We add/delete only changes  and keep unchange as it is

Reply:
I'm just worried about inconsistency because if categoryIds is empty it might be user need, in this case categoryIds could be empty(Or not?). If null = unchanged won't the empty string fields be changed as well?
A little confused @ @

Reply:
When edit the form should populate all the existing categories selected for that product. If no changes, the client need to send all the selected categories, not a empty list